src/main/webapp/scripts/index-script.js

```diff
@@ -1,13 +1,18 @@
-import {updateAccountData, loggedInStatus, signupURL} from './account-info.js';
+import {updateAccountData, loggedInStatus, signupURL, logInOutURL, userInfo}
+  from './account-info.js';
 import {loadNav} from './nav-bar.js';
 
 const dataPromise = updateAccountData();
 
 dataPromise.then(function() {
   loadNav();
+  if (userInfo != null && userInfo.email === 'This') {
```

Comment:
What happens when I have an email address called "This"?? 😄 

Even though that's not a realistic scenario, it's pointing at a real problem: The logic here and the logic in AccountData.java are enforcing a contract of what a "missing user" looks like, and the way you have it built now is not very clear. A contract like that should be effectively self-documenting so it's very difficult to misinterpret in the code.

I wonder if we could use an HTTP code on the response to indicate a missing user. Maybe a 400-error? https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

Reply:
what if i made the email address null? That way, the user cant have email null, because there would be a type mismatch

Reply:
I'd be okay with that